src/Symfony/Bundle/FrameworkBundle/Command/TranslationPullCommand.php

```diff
@@ -0,0 +1,147 @@
+                new InputOption('locales', null, InputOption::VALUE_OPTIONAL | InputOption::VALUE_IS_ARRAY, 'Specify the locales to pull.'),
+                new InputOption('output-format', null, InputOption::VALUE_OPTIONAL, 'Override the default output format.', 'xlf'),
+            ])
+            ->setHelp(<<<'EOF'
+The <info>%command.name%</info> pulls translations from the given provider. Only
+new translations are pulled, existing ones are not overwritten.
+
+You can overwrite existing translations (and remove the missing ones on local side) by using the <comment>--force</> flag:
+
+  <info>php %command.full_name% --force provider</info>
+
+Full example:
+
+  <info>php %command.full_name% provider --force --domains=messages,validators --locales=en</info>
+
+This command pulls all translations associated with the <comment>messages</> and <comment>validators</> domains for the <comment>en</> locale.
+Local translations for the specified domains and locale are deleted if they're not present on the provider and overwritten if it's the case.
+Local translations for others domains and locales are ignored.
+EOF
+            )
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        $provider = $this->providers->get($input->getArgument('provider'));
+        $locales = $input->getOption('locales') ?: $this->enabledLocales;
+        $domains = $input->getOption('domains');
+        $force = $input->getOption('force');
+        $outputFormat = $input->getOption('output-format');
+        $xliffVersion = '1.2';
+
+        switch ($outputFormat) {
+            case 'xlf20': $xliffVersion = '2.0';
+            // no break
+            case 'xlf12': $outputFormat = 'xlf';
+        }
+
+        $writeOptions = [
+            'path' => end($this->transPaths),
+            'xliff_version' => $xliffVersion,
+        ];
+
+        if (!$domains) {
+            $domains = $provider->getDomains();
+        }
+
+        $providerTranslations = $provider->read($domains, $locales);
+
+        if ($force) {
+            foreach ($providerTranslations->getCatalogues() as $catalogue) {
+                $operation = new TargetOperation((new MessageCatalogue($catalogue->getLocale())), $catalogue);
+                $operation->moveMessagesToIntlDomainsIfPossible();
+                $this->writer->write($operation->getResult(), $outputFormat, $writeOptions);
+            }
+
+            $io->success(sprintf('Local translations has been updated from %s (for "%s" locale(s), and "%s" domain(s)).', $provider->getName(), implode(', ', $locales), implode(', ', $domains)));
+
+            return 0;
+        }
+
+        $localTranslations = $this->readLocalTranslations($locales, $domains, $this->transPaths);
+        $translationsToWrite = $providerTranslations->diff($localTranslations);
```

Comment:
I dont get this to work properly.. Are we sure this is correct? I dont get it to fetch remote assets. Loco returns them properly, but they are never added to the local catalogue. 

If I do: 

```php
        $translationsToWrite = $localTranslations->diff($providerTranslations);
```

Then it works, but it writes `messages.en.xlf` instead of `messages+intl-icu.en.xlf`